src/bsoncxx/third_party/CMakeLists.txt

```diff
@@ -12,30 +12,97 @@
+        INSTALL_DIR "${core-install}"
+
         GIT_REPOSITORY https://github.com/mnmlstc/core
         GIT_TAG v1.1.0
+        GIT_SHALLOW TRUE
         LOG_DOWNLOAD ON
-        LOG_CONFIGURE ON
-        LOG_BUILD ON
-        LOG_INSTALL ON
-        LOG_UPDATE ON
-        UPDATE_COMMAND ""
-        CMAKE_ARGS -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER} -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER} -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE} -DBUILD_TESTING=Off -DCMAKE_INSTALL_PREFIX:PATH=${CMAKE_INSTALL_PREFIX}/${BSONCXX_HEADER_INSTALL_DIR}/bsoncxx/third_party/mnmlstc -DINCLUDE_INSTALL_DIR=.
+
+        FETCHCONTENT_UPDATES_DISCONNECTED ON
     )
+    FetchContent_GetProperties (EP_mnmlstc_core)
 
-    ExternalProject_Add_Step(
-        EP_mnmlstc_core
-        fix-includes
-        WORKING_DIRECTORY ${CMAKE_INSTALL_PREFIX}/${BSONCXX_HEADER_INSTALL_DIR}/bsoncxx/third_party/mnmlstc
-        COMMAND find . -type f | xargs perl -pi.bak -e "s|#include <core|#include <bsoncxx/third_party/mnmlstc/core|g"
-        COMMAND find . -name *.bak | xargs rm
-        DEPENDEES install
+    find_package (core NO_DEFAULT_PATH PATHS "${core-install}" QUIET)
+
+    if (core_FOUND AND "$CACHE{INTERNAL_MONGOC_MNMLSTC_CORE_FOUND}")
+        # The mnmlstc/core library is already populated and up-to-date.
+    else ()
+        if (NOT ep_mnmlstc_core_POPULATED)
+            message (STATUS "Downloading mnmlstc/core...")
+            FetchContent_Populate (EP_mnmlstc_core)
+            message (STATUS "Downloading mnmlstc/core... done.")
+        endif ()
+
+        message (STATUS "Configuring mnmlstc/core...")
+        execute_process (
+           COMMAND
+              "${CMAKE_COMMAND}"
+                 "-S" "${core-src}"
+                 "-B" "${core-build}"
+                 "-DCMAKE_INSTALL_PREFIX=${core-install}"
+                 "-DBUILD_TESTING=OFF"
+                 "-DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}"
+                 "-DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}"
+           RESULT_VARIABLE retval
+        )
+        if (NOT "${retval}" STREQUAL "0")
+           message (FATAL_ERROR "execute_process() fatal error: ${retval}")
+        endif ()
+        message (STATUS "Configuring mnmlstc/core... done.")
+
+        message (STATUS "Building mnmlstc/core...")
+        execute_process (
+           COMMAND
+              "${CMAKE_COMMAND}"
+                 --build "${core-build}"
+                 --config "${CMAKE_BUILD_TYPE}"
+           RESULT_VARIABLE retval
+        )
+        if (NOT "${retval}" STREQUAL "0")
+           message (FATAL_ERROR "execute_process() fatal error: ${retval}")
+        endif ()
+        message (STATUS "Building mnmlstc/core... done.")
+
+        message (STATUS "Installing mnmlstc/core...")
+        execute_process (
+           COMMAND
+              "${CMAKE_COMMAND}"
+                 --install "${core-build}"
+                 --config "${CMAKE_BUILD_TYPE}"
+           RESULT_VARIABLE retval
+        )
```

Comment:
The `--install` option was added in [CMake 3.15](https://cmake.org/cmake/help/latest/release/3.15.html):

> The [cmake(1)](https://cmake.org/cmake/help/latest/manual/cmake.1.html#manual:cmake(1)) command gained a new --install option.

Suggest using command compatible with minimum supported CMake version 3.11:

```cmake
        execute_process (
           COMMAND
              "${CMAKE_COMMAND}"
                 --build "${core-build}"
                 --target install
                 --config "${CMAKE_BUILD_TYPE}"
           RESULT_VARIABLE retval
        )
```

Reply:
Resolving since minimum CMake version is bumped to to 3.15.